Mathlib/Data/Polynomial/EraseLead.lean

```diff
@@ -112,27 +112,31 @@ theorem eraseLead_support_card_lt (h : f ≠ 0) : (eraseLead f).support.card < f
   exact card_lt_card (erase_ssubset <| natDegree_mem_support_of_nonzero h)
 #align polynomial.erase_lead_support_card_lt Polynomial.eraseLead_support_card_lt
 
-theorem eraseLead_card_support {c : ℕ} (fc : f.support.card = c) :
-    f.eraseLead.support.card = c - 1 := by
-  by_cases f0 : f = 0
-  · rw [← fc, f0, eraseLead_zero, support_zero, card_empty]
-  · rw [eraseLead_support, card_erase_of_mem (natDegree_mem_support_of_nonzero f0), fc]
-#align polynomial.erase_lead_card_support Polynomial.eraseLead_card_support
+theorem card_support_eraseLead_add_one (h : f ≠ 0) :
+    f.eraseLead.support.card + 1 = f.support.card := by
+  set c := f.support.card with hc
+  cases h₁ : c
+  case zero =>
+    by_contra
+    exact h (card_support_eq_zero.mp h₁)
+  case succ =>
+    rw [eraseLead_support, card_erase_of_mem (natDegree_mem_support_of_nonzero h), ← hc, h₁]
+    rfl
+
+@[simp]
+theorem eraseLead_card_support_one : f.eraseLead.support.card = f.support.card - 1 := by
```

Comment:
```suggestion
theorem card_support_eraseLead_one : f.eraseLead.support.card = f.support.card - 1 := by
```

Reply:
Really? I'd go for

```suggestion
theorem card_support_eraseLead : f.eraseLead.support.card = f.support.card - 1 := by
```

Reply:
Just lemme know which!

Reply:
Sorry, my suggestion had a typo. eric's is correct.

Reply:
Alright, changed to `card_support_eraseLead`.